packages/plugin-core/src/test/perf-test/PerfTesting.test.ts

```diff
@@ -0,0 +1,233 @@
+import {
+  axios,
+  ConfigUtils,
+  DVault,
+  Time,
+  VaultUtils,
+} from "@dendronhq/common-all";
+import { NoteTestUtilsV4 } from "@dendronhq/common-test-utils";
+import { Git } from "@dendronhq/engine-server";
+import { after, describe } from "mocha";
+import os from "os";
+import { performance } from "perf_hooks";
+import { MoveNoteCommand } from "../../commands/MoveNoteCommand";
+import { NoteLookupCommand } from "../../commands/NoteLookupCommand";
+import { RefactorHierarchyCommandV2 } from "../../commands/RefactorHierarchyV2";
+import { ReloadIndexCommand } from "../../commands/ReloadIndex";
+import { ExtensionProvider } from "../../ExtensionProvider";
+import { expect } from "../testUtilsv2";
+import { describeSingleWS } from "../testUtilsV3";
+
+let perflogs: { [key: string]: number } = {};
+
+suite("Performance testing", function () {
+  describe("10000 notes perf testing", () => {
+    [false, true].forEach((enableEngineV3) => {
+      describeSingleWS(
+        `WHEN enableEngineV3 is ${enableEngineV3}`,
+        {
+          timeout: 1e6,
+          preSetupHook: async ({ wsRoot }) => {
+            const git = new Git({
+              localUrl: wsRoot,
+              remoteUrl:
+                "https://github.com/Harshita-mindfire/10000-markdown-files.git",
+            });
+            await git.clone();
+          },
+          modConfigCb: (config) => {
+            const newVault: DVault = {
+              fsPath: "10000-markdown-files",
+              selfContained: true,
+            };
+            const vaults = ConfigUtils.getVaults(config);
+            vaults.push(newVault);
+            config.workspace.vaults = vaults;
+            config.dev!.enableEngineV3 = enableEngineV3;
+            return config;
+          },
+          perflogs,
+        },
+        () => {
+          after(async () => {
+            console.log("******************************************");
+            Object.keys(perflogs).forEach((log) => {
+              console.log(log, "------------->", perflogs[log], "\n");
+            });
+            const headers = {
+              Authorization: `Bearer ${process.env.AIRTABLE_API_KEY}`,
+              "Content-Type": "application/json",
+            };
+            const data = {
+              records: [
+                {
+                  fields: {
+                    date: Time.now().toLocaleString(),
+                    commitHash: process.env.GITHUB_SHA,
+                    githubRef: process.env.GITHUB_REF,
+                    flag: `enableEngineV3: ${enableEngineV3}`,
```

Comment:
maybe instead of calling this `flag` we call it something like `testParameters`? In case in the future, we have different criteria we want to perf test. 

Maybe ok to leave for now, something we change later.